mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/ui/MerchantsFragment.java

```diff
@@ -0,0 +1,136 @@
+
+    @BindView(R.id.inc_state_view)
+    View vStateView;
+    @BindView(R.id.rv_merchants)
+    RecyclerView mRvMerchants;
+    @BindView(R.id.iv_empty_no_transaction_history)
+    ImageView ivTransactionsStateIcon;
+
+    @BindView(R.id.tv_empty_no_transaction_history_title)
+    TextView tvTransactionsStateTitle;
+
+    @BindView(R.id.merchant_fragment_layout)
+    View mMerchantFragmentLayout;
+
+    @BindView(R.id.tv_empty_no_transaction_history_subtitle)
+    TextView tvTransactionsStateSubtitle;
+    private List<SavingsWithAssociations> merchantsList;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ((BaseActivity) getActivity()).getActivityComponent().inject(this);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_merchants, container, false);
+        ButterKnife.bind(this, rootView);
+        showProgressDialog(Constants.PLEASE_WAIT);
+        mPresenter.attachView(this);
+        mMerchantsPresenter.fetchMerchants();
+        return rootView;
+    }
+
+    private void setupUi() {
+        mMerchantsPresenter.checkMerchants();
+    }
+
+    @Override
+    public void showEmptyStateView() {
+        mMerchantFragmentLayout.setVisibility(View.INVISIBLE);
+        if (getActivity() != null) {
+            vStateView.setVisibility(View.VISIBLE);
+            Resources res = getResources();
+            ivTransactionsStateIcon
+                    .setImageDrawable(res.getDrawable(R.drawable.ic_merchants));
+            tvTransactionsStateTitle
+                    .setText(res.getString(R.string.empty_no_merchants_title));
+            tvTransactionsStateSubtitle
+                    .setText(res.getString(R.string.empty_no_merchants_subtitle));
+        }
+    }
+
+    @Override
+    public void hideEmptyStateView() {
+        vStateView.setVisibility(View.GONE);
+        mMerchantFragmentLayout.setVisibility(View.VISIBLE);
+        mRvMerchants.setLayoutManager(new LinearLayoutManager(getContext()));
+        mRvMerchants.setAdapter(mMerchantsAdapter);
+        mRvMerchants.addOnItemTouchListener(new RecyclerItemClickListener(getContext(),
+                new RecyclerItemClickListener.OnItemClickListener() {
+                    @Override
+                    public void onItemClick(View childView, int position) {
+
+                    }
+
+                    @Override
+                    public void onItemLongPress(View childView, int position) {
+
+                    }
+                }));
+    }
+
+    @Override
+    public void setPresenter(MerchantsContract.MerchantsPresenter presenter) {
+        mMerchantsPresenter = presenter;
+    }
+
+    @Override
+    public void listMerchants(List<SavingsWithAssociations> savingsWithAssociationsList) {
```

Comment:
Try to use some more informative names than 'listMerchants'. I have no idea what it does without taking a look into its body.